recipes/default.rb

```diff
@@ -34,26 +34,91 @@
   not_if { node["kagent"]["enabled"] == "false" }
 end
 
-# Generate an API key 
-api_key = ""
+# Generate an API key
+ruby_block 'fetch-api-key' do
```

Comment:
Is it generate an API key or fetch an API key?

Reply:
you can only get an API-key once, so generating and fetching is actually the same thing. changed it to call it `generate-api-key`